session/txninfo/summary.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 PingCAP, Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txninfo
+
+import (
+	"container/list"
+	"encoding/json"
+	"fmt"
+	"hash/fnv"
+	"sync"
+	"time"
+
+	"github.com/pingcap/tidb/types"
+	"github.com/tikv/client-go/v2/oracle"
+)
+
+func digest(digests []string) uint64 {
+	// We use FNV-1a hash to generate the 64bit digest
+	// since 64bit digest use less memory and is FNV-1a is faster than most of other hash algorithms
+	// You can refer to https://softwareengineering.stackexchange.com/questions/49550/which-hashing-algorithm-is-best-for-uniqueness-and-speed
+	hash := fnv.New64a()
+	for _, digest := range digests {
+		hash.Write([]byte(digest))
+	}
+	return hash.Sum64()
+}
+
+type trxSummaryEntry struct {
+	trxDigest uint64
+	digests   []string
+}
+
+type trxSummaries struct {
+	capacity uint
+	size     uint
```

Comment:
Is it necessary to maintain a `size` by ourselves? Can we just use `len(elements)` or `cache.Len()`

Reply:
And why don't we use [the LRU implementation by the Go team](https://github.com/golang/groupcache/blob/master/lru/lru.go)? Though this is simple.

Reply:
> And why don't we use [the LRU implementation by the Go team](https://github.com/golang/groupcache/blob/master/lru/lru.go)? Though this is simple.

IDK whether it is ok to introduce new third party dependency to tidb (there are a few existing implementions of lru in tidb like kvcache.SimpleLRUCache, I don't know whether they don't use this package), but since it's from the Go team I guess it's ok.